### ISSUE TITLE
Unintuitive naming of various fields

### ISSUE THREAD

**Issue by Kixunil** [rust-bitcoin/rust-bitcoin#1824]
I noticed there's a bunch of fields with unintuitive names:

* `Block::txdata` - expected `transactions`
* `OutPoint::vout` - expected `index`
* `block::Header::bits` - expected `target`
* `TxOut::value` - `amount` might be slightly better but I'm not sure

I'd love to rename them if possible.

**Comment by apoelstra** [comment]
`vout` is definitely a standard term. This used to be `previous_index` or something like that and it was constantly confusing. "An outpoint consists of a txid and vout". There is also code formatting benefit in having the word `vout` be the same length as `txid`.

Ditto with `bits`, though this field is basically never touched by users so I wouldn't mind renaming it to `compact_target` or something like that.

`value` is also standard but I'd be willing to change it because it's too general a term, and `txdata` sure is just copied from a Core data structure and I don't think it appears anywhere else.

**Comment by apoelstra** [comment]
Ok, I will defer to the BIP draft (which says "Output Index" rather than "vout", and "amount" rather than "value").

It does not describe blocks or blockheaders yet.

**Comment by Kixunil** [comment]
Oh, cool to have a BIP! Although seems like the BIP is geared towards human language judging from how long some names are.

Notably while I understand that `vout` is used in Core it doesn't seem easy to understand for complete newbies.

**Comment by apoelstra** [comment]
>Notably while I understand that vout is used in Core it doesn't seem easy to understand for complete newbies

Maybe not, but it is used by developers, even when speaking aloud.

It may be worth proposing to extend the BIP to include "programming names" which are single words and somewhat short, for data structure fields.

**Comment by stevenroose** [comment]
If we're gonna be renaming primitive type field names, I was always a little annoyed with `Transaction::input` and `output` instead of `inputs` and `outputs` :) And yeah `Block::txdata` obviously should be `Block::transactions` or `Block::txs` (I prefer the latter personally). I agree with getting rid of `vout` (the v is bitcoind lingo), for something like `index` (I think `OutPoint` already suggests "output" so it might be considered redundant.), but `output_index` also works.

Very slight preference of `amount` over `value`. I agree `bits` isn't the best name, but not really an opinion on another name.

**Comment by tcharding** [comment]
- `txdata` is done already
- `value` -> `amount` done in #4787

#### Other things mentioned in the BIP

- Suggests 'output script' over 'scriptPubkey' (we have `TxOut::script_pubkey`).
- Suggests 'input script' over 'scriptSig' (we have `TxIn::script_sig`)
- Suggests 'leaf script' as a component but does mention 'Tapscript' as a concept. We use `TapScript` 
- Suggests 'lock time' we mention `nLockTime` in docs in `units::lock_time::absolute`

**Comment by apoelstra** [comment]
I'm really torn about 'output script' and 'input script'. I agree these names are much better, and the avoid the idiotic scriptPubKey capitalization that we have to use everywhere we refer to spks. But these terms are almost never used except maybe when you're already talking about an input or output. And even then you might say "scriptpubkey" or "scriptsig" for the avoidance of doubt.

I think it's probably a rename too far.

**Comment by tcharding** [comment]
I tend to agree with you. Feels like these are just warts that are part of Bitcoin. Perhaps we can find a middle ground if we lay out all the pros and cons:

- The names are well known but they are annoying as hell to learn as a dev new to Bitcoin
- The capitalisation is retarded (from a Rust devs perspective)
- The word order is back-to-front, sig-script and pubkey-script would be way cleaner
- The 'intput' in `input_script` is kind of redundant in a `TxIn`, same for 'output' in `output_script` in a `TxOut` 
- rust-bitcoin exists in a wider Bitcoin ecosystem so we do have to make some compromises

And I wrote a more gentle comment with the same content on the BIP: https://github.com/murchandamus/bips/pull/1#issuecomment-3166058033

**Comment by dpc** [comment]
> The 'intput' in `input_script` is kind of redundant in a `TxIn`, same for 'output' in `output_script` in a `TxOut`

Seems like both in the original and all the proposals, the only common word is "script", so just name them both `script` (`TxIn::script` and `TxOut::script`) and be done with it. :D

*I'm not sure myself yet if I'm serious or not.*

**Comment by apoelstra** [comment]
> I'm not sure myself yet if I'm serious or not.

Hah, yeah, this did cross my mind... also unsure whether it's a reasonable thing to do.

**Comment by tcharding** [comment]
Looks like plain old `script` is no good because of the naming thing I mentioned on the bip, for chains of transactions local variables called 'input' can be outputs.

Here is the top few grep results that show the consion

```bash
~/b/rust-bitcoin ❯❯❯ gg '\.script_pubkey(?!\(\))'                                                                                                                                            ✘ 
bitcoin/examples/sign-tx-segwit-v0.rs:65:            &dummy_utxo.script_pubkey,
bitcoin/examples/taproot-psbt.rs:257:            let script_pubkey = ScriptBuf::from_hex_no_length_prefix(input_utxo.script_pubkey)
bitcoin/examples/taproot-psbt.rs:275:            script_pubkey: ScriptBuf::from_hex_no_length_prefix(input.script_pubkey)?,
bitcoin/examples/taproot-psbt.rs:333:            script_pubkey: ScriptBuf::from_hex_no_length_prefix(input_utxo.script_pubkey).unwrap(),
bitcoin/examples/taproot-psbt.rs:509:            let prevout_script_pubkey = input.witness_utxo.as_ref().unwrap().script_pubkey.clone();
bitcoin/examples/taproot-psbt.rs:651:            psbt.inputs[0].witness_utxo.as_ref().unwrap().script_pubkey.clone();
bitcoin/src/bip158.rs:211:                if !output.script_pubkey.is_op_return() {
bitcoin/src/bip158.rs:212:                    self.add_element(output.script_pubkey.as_bytes());
```

**Comment by apoelstra** [comment]
Oh, good catch. When you see something like `input.script_pubkey`, that's meaningful even if it's a little bit sloppy terminology.

Ok, so that kills the "call all the fields `script`" idea in my mind :).

**Comment by dpc** [comment]
I remember expressing preference for "unlock/lock script" over "input/output script" sometime in https://github.com/murchandamus/bips/pull/1, but IIRC at the time I did not have any good arguments. I think the "outputs become inputs, so it gets confusing" is a solid argument. 

Anyway, it's probably good to ponder on it a bit, and report back to broader community, and hopefully help move the needle towards better names. The power of inertia of legacy choices is the reason why Americans are doomed to keep using these terrible socket plugs forever...

**Comment by tcharding** [comment]
I hear some people still type on QWERTY keyboards too, inertia is a hell of a thing. I rekon this bip is a good idea and it can work even with inertia. It doesn't need to be perfect, folk can still say 'vout' occasionally.

**Comment by tcharding** [comment]
Closing because two of the 4 are done and the other two have:

- https://github.com/rust-bitcoin/rust-bitcoin/discussions/4951
- https://github.com/rust-bitcoin/rust-bitcoin/discussions/4952
